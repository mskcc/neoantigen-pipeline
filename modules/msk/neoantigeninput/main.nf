process NEOANTIGENINPUT {
    tag "$meta.id"
    label 'process_single'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigeninputs:1.0.1':
        'docker.io/mskcc/neoantigeninputs:1.0.1' }"

    input:
    tuple val(meta),  path(inputMaf),      path(hlaFile)
    tuple val(meta2), path(phyloWGSsumm),  path(phyloWGSmut),   path(phyloWGSfolder)
    tuple val(meta3), path(mutNetMHCpan),  path(wtNetMHCpan)

    output:
    tuple val(meta), path("*_.json"),                                                  emit: json
    tuple val(meta), path("*.MUT.tsv"), path("*.WT.tsv"),                              emit: netMHCpanreformatted
    path "versions.yml",                                                               emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def id = task.ext.prefix ?: "${meta.id}"
    def patientid = task.ext.cohort ?: "${meta.id}_patient"
    def cohort = task.ext.cohort ?: "${meta.id}_cohort"
    
    """
        ls ${phyloWGSfolder}
        python3 /usr/bin/eval_phyloWGS.py --maf_file ${inputMaf} \
        --summary_file ${phyloWGSsumm} \
        --mutation_file ${phyloWGSmut} \
        --tree_directory ${phyloWGSfolder} \
        --id ${id} --patient_id ${patientid} \
        --cohort ${cohort} --HLA_genes ${hlaFile} \
        --netMHCpan_MUT_input ${mutNetMHCpan} \
        --netMHCpan_WT_input ${wtNetMHCpan}
        ${args}

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            neoantigeninput: \$(echo \$(python3 /usr/bin/eval_phyloWGS.py -v))
        END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def id = task.ext.prefix ?: "${meta.id}"
    def patientid =task.ext.cohort ?: "${meta.id}_patient"
    def cohort =task.ext.cohort ?: "${meta.id}_cohort"
    """
    
        touch ${patientid}_${id}_.json
        touch ${patientid}.MUT.tsv
        touch ${patientid}.WT.tsv
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            neoantigeninput: \$(echo \$(python3 /usr/bin/eval_phyloWGS.py -v))
        END_VERSIONS
    """
}
