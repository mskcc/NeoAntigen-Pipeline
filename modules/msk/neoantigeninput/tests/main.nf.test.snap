{
    "neoantigen": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,cf2994cf0b8fdad159fb8e5191d5618e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1d342cf8b0f1e381e7b0cba890948d23"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,cf2994cf0b8fdad159fb8e5191d5618e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1d342cf8b0f1e381e7b0cba890948d23"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-18T10:02:31.484326"
    }
}